File: src/main/java/com/example/spring/security_jwt/services/RoleService.java
package com.example.spring.security_jwt.services;

import com.example.spring.security_jwt.entities.Role;
import com.example.spring.security_jwt.repositories.RoleRepository;

import org.springframework.stereotype.Service;

import lombok.RequiredArgsConstructor;

@Service
@RequiredArgsConstructor
public class RoleService {
    private final RoleRepository roleRepository;

    public Role getUserRole() {
        return roleRepository.findByName("ROLE_USER")
                .orElseThrow(() -> new RuntimeException("Role not found"));
    }
}

File: src/main/kotlin/com/example/spring/security_jwt/services/RoleService.kt
package com.example.spring.security_jwt.services

import com.example.spring.security_jwt.entities.Role
import com.example.spring.security_jwt.repositories.RoleRepository
import org.springframework.stereotype.Service

@Service
class RoleService(
    private val roleRepository: RoleRepository
) {
    fun getUserRole(): Role {
        return roleRepository.findByName("ROLE_USER")
            .orElseThrow { RuntimeException("Role not found") }
    }
}
